### PR TITLE
Use the right data type for mouse event args

### LINKED ISSUE #11224: Server-Side Blazor Preview 6 breaks onpointerdown event handlers
### Describe the bug
SDK Preview 6 has broken onpointerdown handlers in server-side Blazor

### To Reproduce
1. Using SDK preview 6, create the default server-side Blazor project.
2. In Counter.razor, change `@onclick` to `@onpointerdown` in the button element
3. Run the project and open the debug console in the browser (Chrome, here)
4. Click on the "Click Me" button on the Counter page

### Expected behavior
Counter should increment

Instead, the following error is thrown on the browser console

```
blazor.server.js:8 Uncaught (in promise) Error: System.Text.Json.JsonException: The JSON value could not be converted to System.Int64. Path: $.clientX | LineNumber: 0 | BytePositionInLine: 89.
   at System.Text.Json.ThrowHelper.ThowJsonException(String message, Utf8JsonReader& reader, String path)
   at System.Text.Json.ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(Type propertyType, Utf8JsonReader& reader, String path)
   at System.Text.Json.Serialization.JsonPropertyInfoNotNullable`3.Read(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.Serialization.JsonSerializer.HandleValue(JsonTokenType tokenType, JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& state)
   at System.Text.Json.Serialization.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   at System.Text.Json.Serialization.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
   at System.Text.Json.Serialization.JsonSerializer.ParseCore(String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.Serialization.JsonSerializer.Parse[TValue](String json, JsonSerializerOptions options)
   at Microsoft.AspNetCore.Components.Browser.RendererRegistryEventDispatcher.ParseEventArgsJson(String eventArgsType, String eventArgsJson)
   at Microsoft.AspNetCore.Components.Browser.RendererRegistryEventDispatcher.DispatchEvent(BrowserEventDescriptor eventDescriptor, String eventArgsJson)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.JSInterop.DotNetDispatcher.InvokeSynchronously(String assemblyName, String methodIdentifier, Object targetInstance, String argsJson)
   at Microsoft.JSInterop.DotNetDispatcher.BeginInvoke(String callId, String assemblyName, String methodIdentifier, Int64 dotNetObjectId, String argsJson)
    at endInvokeDotNetFromJS (https://localhost:44322/_framework/blazor.server.js:8:21783)
    at https://localhost:44322/_framework/blazor.server.js:8:21434
    at new Promise (<anonymous>)
    at e.beginInvokeJSFromDotNet (https://localhost:44322/_framework/blazor.server.js:8:21403)
    at https://localhost:44322/_framework/blazor.server.js:1:16653
    at Array.forEach (<anonymous>)
    at e.invokeClientMethod (https://localhost:44322/_framework/blazor.server.js:1:16624)
    at e.processIncomingData (https://localhost:44322/_framework/blazor.server.js:1:14624)
    at e.connection.onreceive (https://localhost:44322/_framework/blazor.server.js:1:11165)
    at WebSocket.i.onmessage (https://localhost:44322/_framework/blazor.server.js:1:30454)
```

Preview 5 did not have this problem.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/AspNetCore/issues/11224

* MDN states that these values were redefined to be double precision floating which is what I see for pointer events when using Chrome v 74. See https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/clientX.
* System.Text.Json does not allow deserializing a `double` into a property backed by a long.
* Commonizing DragEvents since that's the implementation hierarchy - https://developer.mozilla.org/en-US/docs/Web/API/DragEvent
* Added some E2E tests

### CODE REVIEW

File: src/Components/test/E2ETest/Tests/EventTest.cs [hunk 1]
```diff
@@ -117,6 +116,40 @@ public void MouseDownAndMouseUp_CanTrigger()
...
+            var input = Browser.FindElement(By.Id("drag_input"));
+            var target = Browser.FindElement(By.Id("drop"));
+
+            var output = Browser.FindElement(By.Id("output"));
+            Assert.Equal(string.Empty, output.Text);
+
+            var actions = new Actions(Browser).DragAndDrop(input, target);
+
+            actions.Perform();
+            // drop doesn't seem to trigger in Selenium. But it's sufficient to determine "any" drag event works
```
Comment (pranavkm) [inline src/Components/test/E2ETest/Tests/EventTest.cs]: FYI, I can see this working when I run it in a browser (Chrome, Edge, FireFox,), but not using Selenium. I don't see anything obviously wrong with the codez,, but like the comment says if we can verify any drag event works (in this case it's the dragstart) it's sufficient.

**Comment by javiercn** [comment]
I’m not sure if they were typed as long on purpose. Steve might know better.

Regarding selenium, no idea.

**Review by javiercn — APPROVED** [review]
Looks good.

I have a few comments, regarding the use of spread and the change from long to double which I'm not sure if it was done intentionally on the first place for serialization purposes (blazor client-side).

File: src/Components/Browser.JS/src/Rendering/EventForDotNet.ts [hunk 2]
```diff
@@ -91,19 +91,9 @@ export class EventForDotNet<TData extends UIEventArgs> {
 
 function parseDragEvent(event: any) {
   return {
-    type: event.type,
-    detail: event.detail,
+    ...parseMouseEvent(event),
```
[the remark this answers is not in the document]
Comment (pranavkm) [reply src/Components/Browser.JS/src/Rendering/EventForDotNet.ts]: We do something similar with `parseWheelEvent`: https://github.com/aspnet/AspNetCore/blob/81b542a7b03a65003e726781f400b862ca452738/src/Components/Browser.JS/src/Rendering/EventForDotNet.ts#L100-L108 which has the same object hierarchy.

[the remark this answers is not in the document]
Comment (javiercn) [reply src/Components/Browser.JS/src/Rendering/EventForDotNet.ts]: Fair enough

**Review by SteveSandersonMS — APPROVED** [review]
Looks super! Glad you found we could use `double` instead of `long`.

**Pull request merged**